### PR TITLE
BottomSheet migration to react-native-bottom-sheet

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Replace the current `reanimated-bottom-sheet` library with the more up to date and more extensible [react-native-bottom-sheet](https://gorhom.github.io/react-native-bottom-sheet/)

```release-note
NONE
```

### CODE REVIEW

**Comment by matthewbirtch** [comment]
Hey @enahum, it looks like we may have lost some padding on the bottom sheets in this migration. I don't have my computer with me at the moment to check exact numbers, but it looks like we might need 8 more padding on the top of bottom sheets. Seems to be the case across the board (I didn't check tablet yet though)

**Comment by enahum** [comment]
@matthewbirtch basically it had a padding top of 20 for phones while 8 for tablets but I kept it as 8 for everything.. I could make the change back to what it was if needed

**Comment by matthewbirtch** [comment]
> @matthewbirtch basically it had a padding top of 20 for phones while 8 for tablets but I kept it as 8 for everything.. I could make the change back to what it was if needed

Yes please. It needs the extra padding on phone

**Comment by enahum** [comment]
> > @matthewbirtch basically it had a padding top of 20 for phones while 8 for tablets but I kept it as 8 for everything.. I could make the change back to what it was if needed
> 
> Yes please. It needs the extra padding on phone

Changes already applied

**Review by matthewbirtch — APPROVED** [review]
Looks good to me

File: app/screens/post_options/post_options.tsx [hunk 1]
```diff
@@ -63,17 +72,61 @@ const PostOptions = ({
...
+        }
+
+        items.push(bottomSheetSnapPoint(count, ITEM_HEIGHT, bottom) + (canAddReaction ? REACTION_PICKER_HEIGHT + REACTION_PICKER_MARGIN : 0));
+
+        if (shouldShowBindings) {
+            items.push('90%');
+        }
+
+        return items;
+    }, [
```
Comment (jfrerich) [inline app/screens/post_options/post_options.tsx:121]:
```suggestion
    
    const countSnapPoints = (conditions: boolean[]) => {
        return conditions.filter(Boolean).length;
    };

    const snapPoints = useMemo(() => {
        const items: BottomSheetProps['snapPoints'] = [1];
        let count = shouldShowBindings ? 0.5 : 0;
        count += countSnapPoints([
            canReply && sourceScreen !== Screens.THREAD,
            shouldRenderFollow,
            canMarkAsUnread && !isSystemPost,
            canCopyPermalink,
            !isSystemPost,
            Boolean(canCopyText && post.message),
            canPin,
            canEdit,
            canDelete,
        ]);

        items.push(bottomSheetSnapPoint(count, ITEM_HEIGHT, bottom) + (canAddReaction ? REACTION_PICKER_HEIGHT + REACTION_PICKER_MARGIN : 0));

        if (shouldShowBindings) {
            items.push('90%');
        }

        return items;
    }, [
```

Comment (jfrerich) [reply app/screens/post_options/post_options.tsx:121]: Could do something like this if all the `if` statements look ugly and make the conditions more compact to view line by line. 0/5

**Review by jfrerich — APPROVED** [review]
Looks awesome!  I only have one nitpick option for one of the files.

File: app/screens/post_options/post_options.tsx [hunk 2]
```diff
@@ -63,17 +72,61 @@ const PostOptions = ({
...
+        let count = shouldShowBindings ? 0.5 : 0;
+        if (canReply && sourceScreen !== Screens.THREAD) {
+            count++;
+        }
+
+        if (shouldRenderFollow) {
+            count++;
+        }
+
+        if (canMarkAsUnread && !isSystemPost) {
```
Comment (jfrerich) [inline app/screens/post_options/post_options.tsx:86]:
```suggestion
        if (canMarkAsUnread) {
```
Looking in the index file, canMarkAsUnread already accounts for `isSystemPost`
image

File: app/screens/post_options/post_options.tsx [hunk 3]
```diff
@@ -63,17 +72,61 @@ const PostOptions = ({
     const shouldRenderFollow = !(sourceScreen !== Screens.CHANNEL || !thread);
     const shouldShowBindings = bindings.length > 0 && !isSystemPost;
 
-    const snapPoints = [
+    const snapPoints = useMemo(() => {
+        const items: BottomSheetProps['snapPoints'] = [1];
+        let count = shouldShowBindings ? 0.5 : 0;
+        if (canReply && sourceScreen !== Screens.THREAD) {
```
Comment (jfrerich) [inline app/screens/post_options/post_options.tsx:78]:
```suggestion
        if (canReply) {
```
`sourceScreen !== Screens.THREAD` is already accounted for in the index file. 
image

File: app/screens/post_options/post_options.tsx [hunk 4]
```diff
@@ -63,17 +72,61 @@ const PostOptions = ({
...
+
+        if (canCopyPermalink) {
+            count++;
+        }
+
+        if (!isSystemPost) {
+            count++;
+        }
+
+        if (canCopyText && post.message) {
```
Comment (jfrerich) [inline app/screens/post_options/post_options.tsx:98]:
```suggestion
        if (canCopyText) {
```
`post.messaage` is already check for in the `canCopyText` variable assignment.

**Review by jfrerich — COMMENTED** [review]
I added some more comments after looking at the refactor suggestion a bit more. There are some values that are already being checked either in the index file, or in a variable being used.

File: app/screens/post_options/post_options.tsx (hunk 4 above)
Comment (jfrerich) [reply app/screens/post_options/post_options.tsx:98]: image

File: app/screens/post_options/post_options.tsx [hunk 5]
```diff
@@ -63,17 +72,61 @@ const PostOptions = ({
     const shouldRenderFollow = !(sourceScreen !== Screens.CHANNEL || !thread);
     const shouldShowBindings = bindings.length > 0 && !isSystemPost;
 
-    const snapPoints = [
+    const snapPoints = useMemo(() => {
+        const items: BottomSheetProps['snapPoints'] = [1];
```
Comment (jfrerich) [inline app/screens/post_options/post_options.tsx:76]: These could all be pushed into the index file.

`managedConfig?.copyAndPasteProtection !== 'true';` 
* Daniel added an `observeConfigBooleanValue` function that could also be used for this one.

image

Comment (enahum) [reply app/screens/post_options/post_options.tsx:76]: Nope this last one has nothing to do with the config that we get from the server, this is an EMM manage configuration and is retrieved by a hook

File: app/screens/post_options/post_options.tsx (hunk 1 above)
Comment (enahum) [reply app/screens/post_options/post_options.tsx:121]: Thanks for the suggestion, I went with a different approach.

```ts
const optionsCount = [
    canCopyPermalink, canCopyText, canDelete, canEdit,
    canMarkAsUnread, canPin, canReply, !isSystemPost, shouldRenderFollow,
].reduce((acc, v) => {
    return v ? acc + 1 : acc;
}, 0) + (shouldShowBindings ? 0.5 : 0);
```

**Pull request merged**